### PR TITLE
Update docs README.rst to clarify --package-filter usage

### PR DESCRIPTION
**Pull request by RNHTTR**
Make the --package-filter flag a bit more clear, and demonstrate that it can be used for multiple packages.

### CODE REVIEW

**Review by utkarsharma2 — APPROVED** [review]

File: docs/README.rst [hunk 1]
```diff
@@ -53,11 +53,11 @@ or just to run documentation building
 
      breeze build-docs --docs-only
 
-Also, you can only build one documentation via ``--package-filter``.
+Also, you can only build documentation for a subset of packages via ``--package-filter``. For example, to build documentation only for the main Airflow package and the Kubernetes provider package, you can run the following command:
 
 .. code-block:: bash
 
-    breeze build-docs --package-filter <PACKAGE-NAME>
+    breeze build-docs --package-filter apache-airflow --package-filter apache-airflow-providers-cncf-kubernetes
```
Comment (utkarsharma2) [inline docs/README.rst:60]: This example was much needed :)

Comment (utkarsharma2) [reply docs/README.rst:60]: You may want to update here as well - https://github.com/apache/airflow/blob/e6f445129a998eab62d71bd91b4a5f46cd77c1de/dev/breeze/src/airflow_breeze/commands/developer_commands.py#L370

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Just realized @RNHTTR  @utkarsharma2 @Taragolis : We have not finished switching from `--package-filter` to command parameters, it used to be that `--package-fllter had autocompletion and list of packages but before Summit with @amoghrajesh we moved it command option - see #35067

File: dev/breeze/src/airflow_breeze/commands/developer_commands.py [hunk 2]
```diff
@@ -480,7 +480,9 @@ def start_airflow(
     "--package-filter",
     help="List of packages to consider. You can use the full names like apache-airflow-providers-<provider>, "
     "the short hand names or the glob pattern matching the full package name. "
-    "The list of short hand names can be found in --help output",
+    "The list of short hand names can be found in --help output "
+    "Multiple package filters can be applied by passing the flag multiple times. "
+    "For example --package-filter apache-airflow --package-filter apache-airflow-providers-<provider>",
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/developer_commands.py:485]:
```suggestion
    "The list of short hand names can be found in --help output "
    "Multiple package filters can be applied by passing the flag multiple times. "
    "For example --package-filter apache-airflow --package-filter apache-airflow-providers-<provider>. "
    "You can also specify glob-pattern. For example --package-filter '*providers*'." ,
```

**Comment by potiuk** [comment]
Approved - proposed one NIT: improvement (`--package-filter` allows glob pattern matching),  you will akso need to apply static checks to include the description in the help screenshots.

**Review by amoghrajesh — APPROVED** [review]
LGTM +1
Test needs fixing

**Comment by RNHTTR** [comment]
Closed by: #36287

**Pull request closed without merging**